### DISCUSSION TITLE
x-text possible bug: Not working in nested span element

### DISCUSSION THREAD

**Discussion by aroncal** [alpinejs/alpine#3215] [7. Bugs (archived)] [answered]
I found this weird behavior: I have a simple `<div x-data="{foo: {'bar': 'baz'}}">`. When trying to use it for a second time in a span nested inside a `<div>`, it doesn't work and errors with `foo is not defined`

```html
<!DOCTYPE html>
<html>
  <head>
    <title>Alpine x-text possible bug</title>
    <meta charset="UTF-8" />
  </head>

  <body>
    <div x-data="{foo: {'bar': 'baz'}}">
      <!-- 'baz' is printed inside <h3> -->
      <h1 x-text="foo['bar']" />
      <div>
        <!-- ERROR: foo is not defined -->
        <span x-text="foo['bar']" />
      </div>
    </div>

    <script src="src/index.js"></script>
  </body>
</html>
```

I created the followind codesandbox minimal example:
https://codesandbox.io/s/alpine-x-text-bug-lpt86o?file=/index.html

Am I doing something wrong? Thanks.

**Comment by ekwoka** [comment, marked as answer]
Your HTML is invalid

needs to be
`<h1 x-text="foo['bar']"></h1>`

The way you wrote it was replacing the nested divs with the text, and then trying to evaluate the nested expression once it was no longer in the DOM.

Also, you can just do `foo.bar`

**Reply by aroncal** [reply]
I see now. Thank you!
